src/Microsoft.Management.Configuration/Microsoft.Management.Configuration.idl

```diff
@@ -0,0 +1,605 @@
+        // The URI of the documentation for the unit of configuration.
+        Windows.Foundation.Uri UnitDocumentationUri{ get; };
+        // The URI of the icon for the unit of configuration.
+        Windows.Foundation.Uri UnitIconUri{ get; };
+        // The name of the module containing the unit of configuration.
+        String ModuleName{ get; };
+        // The type of the module containing the unit of configuration.
+        String ModuleType{ get; };
+        // The source of the module containing the unit of configuration.
+        String ModuleSource{ get; };
+        // The description of the module containing the unit of configuration.
+        String ModuleDescription{ get; };
+        // The URI of the documentation for the module containing the unit of configuration.
+        Windows.Foundation.Uri ModuleDocumentationUri{ get; };
+        // The URI for the published module containing the unit of configuration.
+        Windows.Foundation.Uri PublishedModuleUri{ get; };
+        // The version of the module containing the unit of configuration.
+        String Version{ get; };
+        // The publishing date of the module containing the unit of configuration.
+        Windows.Foundation.DateTime PublishedDate{ get; };
+        // Whether the module is already present on the system.
+        Boolean IsLocal{ get; };
+        // The author of the module containing the unit of configuration.
+        String Author{ get; };
+        // The publisher of the module containing the unit of configuration.
+        String Publisher{ get; };
+        // The signing certificate chain of the module containing the unit of configuration.
+        Windows.Security.Cryptography.Certificates.CertificateChain SigningCertificateChain{ get; };
+        // The settings information for the unit of configuration.
+        Windows.Foundation.Collections.IVectorView<IConfigurationUnitSettingDetails> Settings{ get; };
+    }
+
+    // Defines how the configuration unit is to be used within the configuration system.
+    [contract(Microsoft.Management.Configuration.Contract, 1)]
+    enum ConfigurationUnitIntent
+    {
+        // The configuration unit will only be used to Test the current system state.
+        Assert,
+        // The configuration unit will only be used to Get the current system state.
+        Inform,
+        // The configuration unit will be used to Set the current system state.
+        // The configuration unit will be used to Test and Get the current system state as part of that process.
+        Apply,
+    };
+
+    // A single unit of configuration.
+    [contract(Microsoft.Management.Configuration.Contract, 1)]
+    runtimeclass ConfigurationUnit
+    {
+        // Creates an empty configuration unit for authoring purposes.
+        ConfigurationUnit();
+
+        // The name of the unit being configured; not a name for this instance.
+        // Can be changed if a mutable unit.
+        String UnitName;

```

Comment:
It is different in that this is the value that we read from the configuration file and will always be present.  The details information requires additional lookups and thus may not be (and it may not even exist because the file has bad values).